crates/vm/src/system/memory/controller/mod.rs

```diff
@@ -346,14 +351,14 @@ impl<F: PrimeField32> MemoryController<F> {
             panic!("Cannot set initial memory after first timestamp");
         }
         let mut offline_memory = self.offline_memory.lock().unwrap();
-        offline_memory.set_initial_memory(memory.clone());
+        offline_memory.set_initial_memory(memory.clone(), self.mem_config);
 
         self.memory = Memory::from_image(memory.clone(), self.mem_config.access_capacity);
 
         match &mut self.interface_chip {
             MemoryInterface::Volatile { .. } => {
                 assert!(
-                    memory.is_empty(),
+                    memory.items().collect::<Vec<_>>().is_empty(),
```

Comment:
Let's add an `is_empty` function?

Also we should probably do `memory.items().next().is_some()` or something rather than collecting the entire memory into a Vec.